### DISCUSSION TITLE
how to add the indicators correctly

### DISCUSSION THREAD

**Discussion by Brianramirez27** [kernc/backtesting.py#458] [General]
Hi the truth, I have read the documentation but I am lost, you can only add indicators from libraries such as ta-lib and tulipy? or you can also from other libraries like ta or pandas ta ??

I am not sure how to add the indicators to the strategy where I should make the calculations in the init?

**Comment by kernc** [comment]
Have you seen [`Strategy.I()`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Strategy.I)?

> `func` is a function that returns the indicator array(s) of same length as Strategy.data. [...] Additional `*args` and `**kwargs` are passed to `func` and can be used for parameters.

**Comment by Brianramirez27** [comment]
This parameter works for which indicator library or only for the specified ones

**Reply by Brianramirez27** [reply]
Suddenly you don't have an example?

**Reply by kernc** [reply]
`func` can be _any_ function (e.g. your own) that simply _returns an array of the same length as data_. Any provided arguments (as [*args and **kwargs](https://realpython.com/python-kwargs-and-args/)) are then simply passed over into `func`, so it only depends on what this function accepts.

See [these FAQ entries that discuss ta and pandas-ta specifically](https://github.com/kernc/backtesting.py/issues?q=is%3Aissue+pandas-ta+label%3Aquestion+).

**Reply by Brianramirez27** [reply]
If I solve it, I have to try the same parameters with other libraries, thank you very much
![pregunta 100](https://user-images.githubusercontent.com/84251018/132712169-7ec0c911-0590-41c3-a47b-712fda105c13.JPG)
